core/optaplanner-constraint-streams-bavet/src/main/java/org/optaplanner/constraint/streams/bavet/uni/UniTuple.java

```diff
@@ -21,7 +21,8 @@
 
 public final class UniTuple<A> implements Tuple {
 
-    public final A factA;
+    // Only a tuple's origin node may modify a fact.
+    public A factA;
```

Comment:
I understand why this exists, and I know there is no other way.
Still, I have to say that I absolutely hate that anyone can just come in and modify the tuples.

Reply:
Yea, but performance begs for it I hope. Let's see how the benchmarks turn out.